src/huak/errors.rs

```diff
@@ -2,57 +2,88 @@ use std::fmt;
 
 use anyhow::Error;
 
-/// Generic CLI Result type.
-pub type CliResult = Result<(), CliError>;
+pub type CliResult<T> = std::result::Result<T, CliError>;
+
+trait BinaryError {}
+
+impl BinaryError for CliErrorType {}
+impl BinaryError for Error {}
+
+#[derive(Debug)]
+pub enum CliErrorType {
+    NotImplemented,
+    MissingVirtualEnv,
+    MissingArguments,
+    UnknownError,
+    IOError,
+    UnknownCommand,
+    DirectoryExists,
+    AnyHowError(anyhow::Error),
+    RuffError(String),
+    PyBlackError(String),
+    PyTest(String),
```

Comment:
What do you think about composing `CliError` in these? Idk if (1) that makes sense or if it's possible. But then we can CliError::new(anyhow, code_from_childprocess).

Reply:
create an enum variant called CliError?

Reply:
ill think about this more.